### PR TITLE
Add load scalarizer optimization.

### PR DESCRIPTION
**Pull request by yjaelex**
This change will scalarize the load instructions;later there is a
"Simplify" pass which will remove the unnecessary load.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 86538e7bf208de1b02f04877b3c157f7720a6793
<details open><summary>Driver commits used in build</summary><ul>

<li>METROHASH: amd-mainline 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 68b57dba33a4d922e8f1ef1b3781c2f659ffbd1c</li>
<li>SPVGEN: dev 2f31d1170e8a12a66168b23235638c4bbc43ecdc</li>
<li>XGL: dev 331558e93794068a786bf699d3fe23bb11bac021</li>
<li>LLVM: amd-vulkan-dev f0a8af5e93d57b890687c7ced43943a0c3a167ca</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 925/925 (100.0%)</li>
<li>Failed: 0/925 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366250/366250 (100.0%)</li>
<li>Failed: 0/366250 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366376/366376 (100.0%)</li>
<li>Failed: 0/366376 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit c3f4bb9d1551d62d987eceff5cd5b36d9557cae5
<details open><summary>Driver commits used in build</summary><ul>

<li>METROHASH: amd-mainline 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 68b57dba33a4d922e8f1ef1b3781c2f659ffbd1c</li>
<li>SPVGEN: dev 2f31d1170e8a12a66168b23235638c4bbc43ecdc</li>
<li>XGL: dev 331558e93794068a786bf699d3fe23bb11bac021</li>
<li>LLVM: amd-vulkan-dev f0a8af5e93d57b890687c7ced43943a0c3a167ca</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 925/925 (100.0%)</li>
<li>Failed: 0/925 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.2.3-cherry-picks)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 366250/366250 (100.0%)</li>
<li>Failed: 0/366250 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 366376/366376 (100.0%)</li>
<li>Failed: 0/366376 (0.0%)</li>
</ul></details>


</ul></details>

File: patch/llpcPatch.cpp [hunk 1]
```diff
@@ -308,6 +308,7 @@ void Patch::AddOptimizationPasses(
         passMgr.add(createSimpleLoopUnrollPass(optLevel));
         passMgr.add(CreatePatchPeepholeOpt());
         passMgr.add(createScalarizerPass());
+        passMgr.add(CreatePatchLoadScalarizer());
```
Comment (jayfoad) [inline patch/llpcPatch.cpp]: This should also be added to the "cl::UseLlvmOpt == true" case around line 364. Or if we are not maintaining that case it should be removed.

Comment (linqun) [reply patch/llpcPatch.cpp]: I agree we need add it to cl::UseLlvmOpt == true

File: patch/llpcPatchLoadScalarizer.cpp [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+    return changed;
+}
+
+// =====================================================================================================================
+// Visits "load" instruction.
+void PatchLoadScalarizer::visitLoadInst(
+    LoadInst& loadInst) // [in] The instruction
+{
+    const uint32_t addrSpace = loadInst.getPointerAddressSpace();
```
Comment (linqun) [inline patch/llpcPatchLoadScalarizer.cpp]: it seems it isn't used, is it expected?

**Pull request merged**